audibot_gazebo/src/AudibotInterfacePlugin.cpp

```diff
@@ -250,6 +257,24 @@ void AudibotInterfacePlugin::feedbackTimerCallback(const ros::TimerEvent& event)
   std_msgs::UInt8 gear_state_msg;
   gear_state_msg.data = gear_cmd_;
   pub_gear_state_.publish(gear_state_msg);
+
+  // added odom publisher.
+  nav_msgs::Odometry odom_msg;
+  odom_msg.header.frame_id = frame_id_;
+  odom_msg.header.stamp = event.current_real;
+  odom_msg.twist.twist = twist_;
+  odom_msg.pose.pose.position.x = world_pose_.Pos().X();
+  odom_msg.pose.pose.position.y = world_pose_.Pos().Y();
+  odom_msg.pose.pose.position.z = world_pose_.Pos().Z();
+  odom_msg.pose.pose.orientation.x = world_pose_.Rot().X();
+  odom_msg.pose.pose.orientation.y = world_pose_.Rot().Y();
+  odom_msg.pose.pose.orientation.z = world_pose_.Rot().Z();
+  odom_msg.pose.pose.orientation.w = world_pose_.Rot().W();
+  pub_odom_.publish(odom_msg);
+
+  std_msgs::Float64 steering;
+  steering.data = current_steering_angle_;
```

Comment:
I agree that the simulator should output the current steering angle because many control strategies can use it and because any car that is controllable by a computer would have that information available. However, production vehicles usually measure the steering wheel angle, not the kinematic steering angle of the tires with the road.

Instead of publishing the kinematic steering angle, please multiply this by `AUDIBOT_STEERING_RATIO` to publish the steering wheel angle instead.